src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -6,45 +6,70 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
-using System.IO.Compression;
 using System.Net.NetworkInformation;
 using System.Runtime.InteropServices;
 using System.Text;
 using System.Threading;
+using Microsoft.Win32;
 using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
 
 namespace System.Net.Http
 {
     internal sealed class HttpWindowsProxy : IMultiWebProxy, IDisposable
     {
-        private readonly MultiProxy _insecureProxy;    // URI of the http system proxy if set
-        private readonly MultiProxy _secureProxy;      // URI of the https system proxy if set
-        private readonly FailedProxyCache _failedProxies = new FailedProxyCache();
-        private readonly List<string>? _bypass;         // list of domains not to proxy
-        private readonly bool _bypassLocal;    // we should bypass domain considered local
-        private readonly List<IPAddress>? _localIp;
+        private readonly RegistryKey? _internetSettingsRegistry = Registry.CurrentUser?.OpenSubKey("Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings");
+        private MultiProxy _insecureProxy;      // URI of the http system proxy if set
+        private  MultiProxy _secureProxy;       // URI of the https system proxy if set
+        private FailedProxyCache _failedProxies = new FailedProxyCache();
+        private List<string>? _bypass;          // list of domains not to proxy
+        private List<IPAddress>? _localIp;
         private ICredentials? _credentials;
-        private readonly WinInetProxyHelper _proxyHelper;
+        private WinInetProxyHelper _proxyHelper;
         private SafeWinHttpHandle? _sessionHandle;
         private bool _disposed;
+        private EventWaitHandle _waitHandle = new EventWaitHandle(false, EventResetMode.AutoReset);
+        private const int _registrationFlags = Interop.Advapi32.REG_NOTIFY_CHANGE_NAME | Interop.Advapi32.REG_NOTIFY_CHANGE_LAST_SET | Interop.Advapi32.REG_NOTIFY_CHANGE_ATTRIBUTES | Interop.Advapi32.REG_NOTIFY_THREAD_AGNOSTIC;
+        private RegisteredWaitHandle? _registeredWaitHandle;
 
         public static bool TryCreate([NotNullWhen(true)] out IWebProxy? proxy)
```

Comment:
Does TryCreate still need to exist if it's just the same as the public ctor?

Reply:
it does not. I was thinking about it originally and I decided to keep it. Since you mentioned I changed it to use public constructor as it better matches reality.